graphistry/umap_utils.py

```diff
@@ -491,6 +495,11 @@ def umap(
 
             logger.debug("umap X_: %s", X_)
             logger.debug("umap y_: %s", y_)
+            logger.debug("data is type :: %s", (type(X_)))
+            if isinstance(X_, pd.DataFrame):
+                index_to_nodes_dict = dict(zip(range(len(nodes)), nodes))
+            elif 'cudf.core.dataframe' in str(getmodule(X_)):
+                index_to_nodes_dict = nodes
```

Comment:
when cudf, is `index_to_nodes_dict = nodes` still a dict? @dcolinmorgan

Reply:
how about... always make it a df, and just varies whether cpu vs gpu?

Reply:
yeah its already a dict from cudf